Std/Data/String/Lemmas.lean

```diff
@@ -33,6 +33,16 @@ theorem ext_iff {s₁ s₂ : String} : s₁ = s₂ ↔ s₁.data = s₂.data :=
   rw [push, mk_length, List.length_append, List.length_singleton, Nat.succ.injEq]
   rfl
 
+@[simp] theorem length_pushn (c : Char) (n : Nat) : (pushn s c n).length = s.length + n := by
+  induction n with
+  | zero => simp [pushn, Nat.repeat]
+  | succ k h =>
+    simp [pushn, Nat.repeat, Nat.add_succ]
+    exact h
```

Comment:
```suggestion
  unfold pushn; induction n <;> simp [Nat.repeat, Nat.add_succ, *]
```

Reply:
Yeah that's way better, I wasn't really trying to golf that proof in the first place